### PR TITLE
DAOS-2617 documentation: update release notes and API

### PR DESCRIPTION
**Pull request by cdurf1**
Skip-build: true
Skip-test: true

Signed-off-by: craig <craig.durfey@intel.com>

### CODE REVIEW

**Comment by cdurf1** [comment]
Only Release notes require review. The other files are for the API

**Review by johannlombardi — CHANGES_REQUESTED** [review]
all the code under doc/html should not go in the repository.

File: doc/release/release_notes_v1_2.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
+# DAOS Version 1.2 Release Notes
+
+We are pleased to announce the release of DAOS version 1.2, advancing on a newly created high-performance object store that is ultimately
+shifting the HPC and big data paradigm. 
+
+This release is targeted towards benchmarking, partner integration, evaluation,
+and brings support for the following new features:
+
+-  End-to-end Data Integrity
```
Comment (brianjmurrell) [inline doc/release/release_notes_v1_2.md:9]: Extra space here:
```suggestion
- End-to-end Data Integrity
```

File: doc/release/release_notes_v1_2.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+- Per Container ACL
+- Improved Control Plane
+- Replication and Self Healing
+- Erasure Code (Preview)
+- Off-line Re-integration
+-  HDF5 with Async
```
Comment (brianjmurrell) [inline doc/release/release_notes_v1_2.md:15]: Ditto.

```suggestion
- HDF5 with Async
```

File: doc/release/release_notes_v1_2.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+be posted in the [DAOS mailing list](https://daos.groups.io/g/daos).
+
+## Build Prerequisites
+To build DAOS and its dependencies, several software packages must be installed
+on the system. This includes scons, libuuid, cmocka, ipmctl, and several other
+packages usually available on all the Linux distributions. Moreover, a Go
+version of at least 1.10 is required.
+
+[CentOS](https://github.com/daos-stack/daos/blob/master/utils/docker/Dockerfile.centos.7#L53-L76)
+is currently the only supported Linux distribution. An exhaustive list of
```
Comment (brianjmurrell) [inline doc/release/release_notes_v1_2.md:46]: We are not supporting Leap/SLES 15 yet, officially?

File: doc/release/release_notes_v1_2.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+For testing purposes, SCM can be emulated with DRAM by mounting a tmpfs
+filesystem, and NVMe SSDs can be also emulated with DRAM or a loopback
+file. Any data written using a tmpfs will not be persistent across a reboot.
+
+
+## DAOS Testing
+DAOS 1.2 validation efforts were focused on anticipated initial use cases. Testing has been completed in the following areas:
+
+- Testing has been performed on CentOS 7.9, OpenSuse 15.2,with CentOS  being used in the majority of the test cycles. Smaller scale testing has been done on OpenSuse 15.2, and no testing performed on Ubuntu and MOFED 5.1x
```
Comment (brianjmurrell) [inline doc/release/release_notes_v1_2.md:100]:
```suggestion
- Testing has been performed on CentOS 7.9, OpenSuse 15.2, with CentOS  being used in the majority of the test cycles. Smaller scale testing has been done on OpenSuse 15.2, and no testing performed on Ubuntu and MOFED 5.1x
```

File: doc/release/release_notes_v1_2.md [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+With a configuration of only Intel(R) Optane Pmem for storage in a DAOS pool
+(no SSDs), a known limitation currently exists on the write/update performance
+with bulk transfers to the SCM with all network providers which limits the BW
+per storage server.
+
+### Aggregation Issue
+The aggregation service is only activated after the system has idled for more
+than 5 seconds, so the aggregation is always suspended if there are sustained
+I/O requests from application. As a result, storage node will run out of SCM
+space even there is plenty of free space in SSD
```
Comment (brianjmurrell) [inline doc/release/release_notes_v1_2.md:356]: This seems to be a repeat of lines 298-326.

**Comment by cdurf1** [comment]
Closing this request to delete html files

File: doc/release/release_notes_v1_2.md [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+[More information]( https://daos-stack.github.io/admin/pool_operations/#target-exclusion-and-self-healing)
+
+### Erasure Code (Preview)
+
+This release implements Reed Solomon based EC for data protection of DAOS, it supports EC data recovery for storage target failure, and data migration while extending the storage pool. Main features of DAOS EC include:
+
+ - Reed-Solomon based EC support for I/O. DAOS has two value types: array value for large streaming I/O; single value for Key-value store. DAOS EC supports both array value and single value. An array value always has fixed stripe size and cell size, whereas a single value has dynamic stripe size to provide storage efficiency of application data with arbitrary size. This feature also provides a set of predefined EC object classes as its user interface. The user can choose the appropriate object class for their data based on their requirements and restrictions. For example, capacity and bandwidth overhead, fault tolerance level, etc.
```
Comment (skirvan44) [inline doc/release/release_notes_v1_2.md:187]: This feature also provides a set of predefined EC object classes as its user interface.  Perhaps instead:

The EC feature is activated by utilizing one of the predefined EC object classes.

File: doc/release/release_notes_v1_2.md [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+ - Data recovery for EC protected object.  DAOS server failure may result in loss of access to part of the EC protected data. If number of failures is equal or less than the EC redundancy factor, DAOS client can retrieve a quorum of surviving cells and reconstruct the inaccessible data for application that is trying to read, whereas DAOS server can reconstruct lost data in the background and recover data redundancy to the original level. 
+
+
+
+### Off-line Re-integration
+
+In DAOS, the excluded targets can be reintegrated back into the pool by dmg pool command, though the pool should not be connected (no I/O) during reintegration, otherwise application failure or even data corruption might be observed, which will be fixed in the later version. The status of the reintegration can also be checked through dmg pool query, similar as rebuild. During reintegration, all old data on the reintegrated target will be erased. Once the target is reintegrated, it operates the same as normal active targets.
```
Comment (skirvan44) [inline doc/release/release_notes_v1_2.md:195]: Could we be explicit about the meaning of target here?

File: doc/release/release_notes_v1_2.md [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+If you must evaluate DAOS on an OPA fabric, we recommend using IP over
+fabric/sockets for evaluation.  For benchmarking, PSM2 over OPA offers a
+better picture of DAOS performance, but the PSM2 provider is not fully
+functional. For a discussion of issues and options, contact us via the
+[mailing list](https://daos.groups.io/g/daos).
+
+### Replication and Self-healing
+While replication and self-healing features have been fully implemented,
+those features haven't been thoroughly tested in 1.2. As a result, they are
+supported as preview features and will be fully supported in the next release.
```
Comment (skirvan44) [inline doc/release/release_notes_v1_2.md:309]: Is this left over from 1.0?  I believe we are claiming support for replication & self healing

File: doc/release/release_notes_v1_2.md [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+section of the administration guide for more information.
+
+### Limited bandwidth to SCM with some update/write operations
+With a configuration of only Intel(R) Optane Pmem for storage in a DAOS pool
+(no SSDs), a known limitation currently exists on the write/update performance
+with bulk transfers to the SCM with all network providers which limits the BW
+per storage server.
+
+### Aggregation Issue
+The aggregation service is only activated after the system has idled for more
```
Comment (skirvan44) [inline doc/release/release_notes_v1_2.md:322]: ditto here, left over from 1.0?

**Pull request closed without merging**